cmd/virt-launcher/virt-launcher.go

```diff
@@ -610,16 +617,28 @@ func RemoveContents(dir string) error {
 
 func terminateIstioProxy() {
 	if istioProxyPresent() {
-		resp, err := http.Post(fmt.Sprintf("http://localhost:%d/quitquitquit", infraconfigurators.EnvoyMergedPrometheusTelemetryPort), "", nil)
-		if err != nil || resp.StatusCode != http.StatusOK {
-			log.Log.Error("Failed to request Istio proxy termination")
+		isRetriable := func(err error) bool {
+			if net.IsConnectionReset(err) || net.IsConnectionRefused(err) || errors2.IsServiceUnavailable(err) {
+				return true
+			}
+			return false
 		}
+		err := retry.OnError(retry.DefaultBackoff, isRetriable, func() error {
+			resp, err := httpClient.Post(fmt.Sprintf("http://localhost:%d/quitquitquit", infraconfigurators.EnvoyMergedPrometheusTelemetryPort), "", nil)
+			if err != nil || resp.StatusCode != http.StatusOK {
+				log.Log.Error("failed to request istio-proxy termination, retrying...")
+				return err
+			}
+			return nil
+		})
+		log.Log.Reason(err).Error("all attempts to terminate istio-proxy failed")
 	}
 }
 
 func istioProxyPresent() bool {
-	resp, err := http.Get(fmt.Sprintf("http://localhost:%d/healthz/ready", infraconfigurators.EnvoyHealthCheckPort))
+	resp, err := httpClient.Get(fmt.Sprintf("http://localhost:%d/healthz/ready", infraconfigurators.EnvoyHealthCheckPort))
```

Comment:
Is it relevant to backoff/retry the GET too?

Reply:
Added retry for cases where we get connectionReset or connectionRefused.
It's not relevant to check for ServiceUnavailable, as that would reveal if the response is or is not from the envoy.

Reply:
As for connectionRefused, it's actually not too relevant too, since this is a good indication that the port is closed --> proxy not present. But there's no real harm in retrying, since connectionRefused should fail fairly quick, and the backoff policy used specifies 4-5 quick retries.